Add VelocityPolygon docs

kaichie opened the pull request:
For https://github.com/ros-planning/navigation2/pull/3708

SteveMacenski commented on configuration/packages/collision_monitor/configuring-collision-monitor-node.rst:
```diff
@@ -364,6 +365,105 @@ Polygons parameters
+  ============== =============================
+
+  Description:
+    Polygon vertexes, listed in ``"[[p1.x, p1.y], [p2.x, p2.y], [p3.x, p3.y], ...]"`` format (e.g. ``"[[0.5, 0.25], [0.5, -0.25], [0.0, -0.25], [0.0, 0.25]]"`` for the square in the front). Used for ``polygon`` type. Minimum 3 points for a triangle polygon. Causes an error, if not specified. Applicable for the sub polygon in `velocity_polygon` type only.
```
This and below: it’s unclear that these are for velocity polygons when inline like this. Please group these under a subsection in this param section for velocity polygon specific params

SteveMacenski replied:
And param above too.

SteveMacenski commented on configuration/packages/collision_monitor/configuring-collision-monitor-node.rst:
```diff
@@ -364,6 +365,105 @@ Polygons parameters
+    Description:
+      Maximum angular velocity for the sub polygon. Causes an error, if not specified. Applicable for the sub polygon in `velocity_polygon` type only.
+
+:``<polygon_name>.<sub_polygon_name>``.direction_start_angle:
```
Put those diagrams explaining the direction semantics here too, not just in the tutorial

SteveMacenski commented:
I think also this comment wasn't documented:

> Perhaps this is just a documentation item that users should set specific over general polygons in the list since the first that applies is used (so backups at the end)

Missing docs that the first polygon, if many cover a range, is used first. That should be addressed in the tutorial, configuration guide, and readme

SteveMacenski commented on configuration/packages/collision_monitor/configuring-collision-monitor-node.rst:
```diff
@@ -364,6 +365,105 @@ Polygons parameters
+  ============== =============================
+
+  Description:
+    Whether to use holonomic or non-holonomic robot model for collision prediction. Applicable for `velocity_polygon` type only.
```
Refing the tutorial with the direction stuff would be important here

SteveMacenski commented on configuration/packages/collision_monitor/configuring-collision-monitor-node.rst:
```diff
@@ -364,6 +365,108 @@ Polygons parameters
 
+VelocityPolygon parameters
+==========================
+
```
Add a line here that all previous parameters apply, in addition to the following unique parameters for velocity polygons

SteveMacenski commented on migration/Iron.rst:
```diff
@@ -5,6 +5,12 @@ Iron to Jazzy
+Add VelocityPolygon in Collision Monitor
+****************************************
+
+`PR #3708 <https://github.com/ros-planning/navigation2/pull/3708>`_ adds ``VelocityPolgon`` type in Collision Monitor. This allows the user to setup multiple polygons to cover the range of the robot's velocity limits. For example, the user can configure different polygons for rotation, moving forward, or moving backward. The Collision Monitor will check the robot's velocity against each sub polyg ...
```
Add `:ref:` to the tutorial update on it

kaichie commented:
> @kaichie look at the rendering: there's something wrong with the fformatting or your new section https://output.circle-artifacts.com/output/job/2288c871-ef24-4297-8327-cddd9214fe45/artifacts/0/html/configuration/packages/collision_monitor/configuring-collision-monitor-node.html

I could not see anything obvious that is wrong with the table formatting. Looks like it is affected by the length of the parameter name on the left. Not familiar with this but will continue looking later, appreciate any tips here.

SteveMacenski commented:
I have no particular suggestion. given that this is already a problem, I don't expect you to fix it (though that would be ideal), I would however expect you to make it less bad. It seems like reducing the `<velocity_polygon_name>.<sub_polygon_name>` length would do alot of good. Perhaps `<vel_poly>.<subpoly>`

If that's not quite enough, seeing what formatting is not properly IDing these tables max width

kaichie commented:
Attempted to fix it in this [commit](https://github.com/ros-planning/navigation.ros.org/pull/510/commits/bdacb7d5043a40f3f5d6c185fb18c2c79b517b84), looks okay on the [collision monitor page](https://output.circle-artifacts.com/output/job/d62c3141-27b6-4233-a1eb-feeac636f852/artifacts/0/html/configuration/packages/collision_monitor/configuring-collision-monitor-node.html), but causes the table on other pages to look a bit funny, for example, [here](https://output.circle-artifacts.com/output/job/d62c3141-27b6-4233-a1eb-feeac636f852/artifacts/0/html/configuration/packages/bt-plugins/actions/Wait.html#input-ports). Reverted and updated the parameter length to `<vel_poly>.<subpoly>`.

SteveMacenski commented:
OK, works for me for now
